### PR TITLE
add drainer notify test for pump node

### PR DESCRIPTION
**Pull request by lichunzhu**
### What problem does this PR solve? 
[TOOL-1056](https://internal.pingcap.net/jira/browse/TOOL-1056)

### What is changed and how it works?
Mock a fake grpc drainer to be notified. Mocked drainer will occupy port 8249 for a while.

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

Side effects

Related changes

### CODE REVIEW

File: pump/node_test.go [hunk 1]
```diff
@@ -168,3 +171,48 @@ func (s *heartbeatSuite) TestShouldUpdateStatus(c *C) {
...
+}
+
+func (s *notifyDrainerSuite) TestNotifyDrainer(c *C) {
+	host := "127.0.0.1:8249"
+	lis, err := net.Listen("tcp", host)
+	c.Assert(err, IsNil)
+	gs := grpc.NewServer()
+	mockDrainerServerImpl := &mockDrainerServer{notified: false}
+	binlog.RegisterCisternServer(gs, mockDrainerServerImpl)
+	// defer gs.Stop()
```
Comment (suzaku) [inline pump/node_test.go]: Why is this line commented?

File: pump/node_test.go [hunk 2]
```diff
@@ -168,3 +171,48 @@ func (s *heartbeatSuite) TestShouldUpdateStatus(c *C) {
...
+		},
+	}
+	drainerNodeStatus := &node.Status{
+		NodeID: "pump_notify_test",
+		State:  node.Online,
+		Addr:   host,
+	}
+	err = registry.UpdateNode(context.Background(), "drainers", drainerNodeStatus)
+	c.Assert(err, IsNil)
+	err = pNode.Notify(context.Background())
```
Comment (suzaku) [inline pump/node_test.go]: There's a basic structure in most tests:

1. Setup dependencies
1. Run the target against some input
1. Validate the output

We can arrange our test in three paragraphs and insert blank lines between them to make it clearer.

For each paragraph, we may add comment to explain what's being setup if it's not clear from the context.

In this case, the setup step is quite a lot of work, you may want to make it simpler:

1. Is it possible to extract some functions?
1. Is the etcd setup necessary? Can we replace it with a mock?

File: pump/node_test.go (hunk 1 above)
Comment (lichunzhu) [reply pump/node_test.go]: Revised.

File: pump/node_test.go (hunk 2 above)
Comment (lichunzhu) [reply pump/node_test.go]: 1. I simply the start of mock drainer as a function of `mockDrainerServer`.
2. IMO, the price of mock etcd setup is much higher than current implementation. Especially in function `pump/node.go:176:notifyDrainer` many varibles are very hard to mock.

**Comment by july2993** [comment]
@leoppro PTAL

**Pull request merged**